homeworks/07-mars-viewer/src/store/reducers/fetchSolsReducer.ts

```diff
@@ -0,0 +1,32 @@
+import { types } from '../actions/actionsTypes';
+import { FetchSols, Sols } from '../actions/appActions';
+
+interface InitialState {sols: Sols, loading: boolean}
+
+const initialState: InitialState = {
+  sols: [],
+  loading: false,
+};
+
+export const fetchSolsReducer = (state = initialState, action: FetchSols): InitialState => {
+  const { type, payload } = action;
+  switch (type) {
+    case types.FetchSolsStart:
+      return {
+        ...state,
+        loading: payload.loading,
+      };
+    case types.FetchSolsSuccess:
+      return {
+        ...state,
+        loading: payload.loading,
+        sols: payload.sols,
```

Comment:
this leads to runtime error and app crash, steps to reproduce:
1. load first sol
2. add some image to favourites
3. load another sol and go to favourites and get
`TypeError: Cannot read property 'img_src' of undefined`

This happens because data for new sol is not being concatenated with data from other sols